scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -244,12 +297,12 @@ static bool is_task_cpu_available(struct task_struct *p)
 	struct task_ctx *tctx;
 
 	/*
-	 * Always dispatch per-CPU kthread on the same CPU, bypassing the
-	 * user-space scheduler (in this way we can to prioritize critical
-	 * kernel threads that may potentially slow down the entire system if
-	 * they are blocked for too long).
+	 * Always dispatch kthread on the same CPU, bypassing the user-space
+	 * scheduler (in this way we can to prioritize critical kernel threads
+	 * that may potentially slow down the entire system if they are blocked
+	 * for too long).
 	 */
-	if (is_kthread(p) && p->nr_cpus_allowed == 1)
+	if (is_kthread(p))
```

Comment:
Not an immediate blocker but this might lead to not-so-great behaviors for things like unbound workqueue workers that can consume a lot of CPU cycles (e.g. encryption workers). They actually need scheduling.

Reply:
> Not an immediate blocker but this might lead to not-so-great behaviors for things like unbound workqueue workers that can consume a lot of CPU cycles (e.g. encryption workers). They actually need scheduling.

Good point! Yes, in that case they definitely need scheduling, I'll revert that in the next set of changes.